bio_files_processor.py

```diff
@@ -0,0 +1,135 @@
+import os
+import re
+from dataclasses import dataclass
+
+
+def convert_multiline_fasta_to_oneline(
+        input_fasta: str, output_fasta: str = None) -> None:
+    """
+    Converts a multiline FASTA file to an oneline FASTA file.
+
+    Parameters:
+    - input_fasta: Path to the input multiline FASTA file
+    with extension '.fasta' or '.fa'
+    - output_fasta: Path to the output oneline FASTA file
+    (default: input_name + '_oneline' + ext)
+
+    Returns:
+    - None
+
+    Steps:
+    1. Check if input file is a FASTA file
+    2. Set output file name if not provided
+    3. Check if output file is already exists
+    4. Read input file line by line and convert to oneline format
+    5. Write the oneline entries to output file
+    """
+    # check if `input_fasta` a fasta file:
+    pattern = r'^(.+?)(\.fasta|\.fa)$'
+    match = re.match(pattern, input_fasta)
+    if match:
+        input_name = match.group(1)
+        ext = match.group(2)
+    else:
+        raise ValueError(f"File '{input_fasta}' is not a fasta file")
+    # check if `output_fasta` is passed and set it if None:
+    if output_fasta is None:
+        output_fasta = input_name + '_oneline' + ext
+    # check if 'output_fasta' exists:
+    if os.path.isfile(output_fasta):
+        raise FileExistsError(f"File '{output_fasta}' is already exists")
+
+    # handle input_fasta line-by-line:
+    with open(input_fasta, 'r') as f:
+        l = f.readline().strip()
```

Comment:
Однобуквенные переменные все же не очень хорошо)
```suggestion
        line = f.readline().strip()
```